### PR TITLE
Add: [CMake] targets to generate documentation

### PR DESCRIPTION
**Pull request by glx22**
This should simplify doc generation. I also removed some outdated doxygen settings.

### CODE REVIEW

File: src/script/api/Doxyfile_GS.in [hunk 1]
```diff
@@ -10,8 +10,8 @@
 #---------------------------------------------------------------------------
 DOXYFILE_ENCODING      = UTF-8
 PROJECT_NAME           = "OpenTTD GameScript API"
-PROJECT_NUMBER         = $(VERSION)
-OUTPUT_DIRECTORY       = ../../../docs/gamedocs/
+PROJECT_NUMBER         = ${REV_VERSION}
+OUTPUT_DIRECTORY       = ${CPACK_BINARY_DIR}/docs/gamedocs/
```
Comment (TrueBrain) [inline src/script/api/Doxyfile_GS.in]: `gamedocs` always has been an odd name, as it is wrong in so many ways. Possibly a good moment to rename it?

`gs-api`? As that is what it is called here?
Maybe also change AI to be `ai-api` too while at it? :D

File: src/script/api/Doxyfile_GS.in [hunk 2]
```diff
@@ -211,15 +207,13 @@ SKIP_FUNCTION_MACROS   = YES
 # Configuration::additions related to external references
 #---------------------------------------------------------------------------
 TAGFILES               =
-GENERATE_TAGFILE       = ../../../objs/openttd_nogo.tag
+GENERATE_TAGFILE       = ${CPACK_BINARY_DIR}/docs/openttd_nogo.tag
```
Comment (TrueBrain) [inline src/script/api/Doxyfile_GS.in]: lol, and here it is "nogo" ... I must have been drunk when making this :D Please also change ;)

File: src/script/api/Doxyfile_AI.in [hunk 3]
```diff
@@ -91,7 +90,7 @@ WARN_LOGFILE           =
 #---------------------------------------------------------------------------
 # configuration options related to the input files
 #---------------------------------------------------------------------------
-INPUT                  = . $(GENERATED_API_DIR)
+INPUT                  = . ${FIND_VERSION_BINARY_DIR}/script/api
```
Comment (TrueBrain) [inline src/script/api/Doxyfile_AI.in:93]: `FIND_VERSION`? Shouldn't this be `CMAKE_SOURCE_DIR`?

File: src/script/api/Doxyfile_AI.in [hunk 4]
```diff
@@ -10,8 +10,8 @@
 #---------------------------------------------------------------------------
 DOXYFILE_ENCODING      = UTF-8
 PROJECT_NAME           = "OpenTTD AI API"
-PROJECT_NUMBER         = $(VERSION)
-OUTPUT_DIRECTORY       = ../../../docs/aidocs/
+PROJECT_NUMBER         = ${REV_VERSION}
+OUTPUT_DIRECTORY       = ${CPACK_BINARY_DIR}/docs/aidocs/
```
Comment (TrueBrain) [inline src/script/api/Doxyfile_AI.in]: `CPACK_BINARY_DIR`? Not `CMAKE_BINARY_DIR`?

**Review by TrueBrain — CHANGES_REQUESTED** [review]

File: src/script/api/Doxyfile_AI.in (hunk 4 above)
Comment (glx22) [reply src/script/api/Doxyfile_AI.in]: `CMAKE_BINARY_DIR` is not known inside `FindVersion.cmake` so I used the available variables passed by the target command

Edit: see https://github.com/OpenTTD/OpenTTD/blob/7f5afbfdf5ac288da188276430c30b3ed0af99f8/CMakeLists.txt#L121-L128

File: src/script/api/Doxyfile_AI.in (hunk 3 above)
Comment (glx22) [reply src/script/api/Doxyfile_AI.in:93]: `.` is actually `CMAKE_CURRENT_SOURCE_DIR` (via `WORKING_DIRECTORY` in the target). And we also need generated files (`script_window.hpp` mainly) and `FindVersion.cmake` gets it via target command

Edit: see https://github.com/OpenTTD/OpenTTD/blob/7f5afbfdf5ac288da188276430c30b3ed0af99f8/CMakeLists.txt#L121-L128

File: .github/workflows/ci-build.yml [hunk 5]
```diff
@@ -61,6 +61,30 @@ jobs:
...
+
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: CMake
+      run: mkdir build && cd build && cmake .. -DOPTION_DEDICATED=ON
+
+    - name: Build
+      run: cd build && make docs
+
```
Comment (TrueBrain) [inline .github/workflows/ci-build.yml]: I did some thinking about this .. if we do not plan to fix the warnings for any of the three targets, running it in the CI is just a waste of CPU honestly :P So I guess we have to pick either one or the other .. either we plan to fix warnings in `ai-api` and `gs-api`, or we should not run it in the CI :) What do you think?

Comment (glx22) [reply .github/workflows/ci-build.yml]: I agree, a nightly run should be enough.

**Review by TrueBrain — APPROVED** [review]
LGTM

**Pull request merged**